### DISCUSSION TITLE
Issue starting Etherpad as root

### DISCUSSION THREAD

**Discussion by mkamranmazhar** [ether/etherpad-lite#6395] [Q&A] [answered]
Hi everyone,

I'm encountering an issue when trying to start Etherpad as root on my server. Here's the problem I'm facing:

I've followed the instructions to clone Etherpad from GitHub and run it using the `sudo bin/run.sh --root` command. However, I'm getting an error related to unrecognized file extensions, specifically with TypeScript files.

Here's the error message I'm receiving:

`root@vmi1855517:/www/wwwroot/tools/etherpad-lite# sudo bin/run.sh --root
Copy the settings template to settings.json...
Installing dependencies...
Installing dev dependencies with pnpm
Scope: all 6 workspace projects
Lockfile is up to date, resolution step is skipped
.                                        | +877 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 877, reused 877, downloaded 0, added 877, done
Done in 4.3s
Clearing minified cache...
Creating the admin UI...

> admin@2.0.3 build /www/wwwroot/tools/etherpad-lite/admin
> tsc && vite build

vite v5.2.9 building for production...
✓ 1641 modules transformed.
computing gzip size (2)...[vite-plugin-static-copy] Copied 1 items.
../src/templates/admin/index.html                   0.49 kB │ gzip:   0.30 kB
../src/templates/admin/assets/index-E-lmtrZj.css   10.20 kB │ gzip:   3.02 kB
../src/templates/admin/assets/index-DIlmNsYJ.js   407.69 kB │ gzip: 130.80 kB
✓ built in 5.82s

> ui@0.0.0 build /www/wwwroot/tools/etherpad-lite/ui
> tsc && vite build

vite v5.2.9 building for production...
✓ 6 modules transformed.
../src/static/oidc/consent.html               1.01 kB │ gzip: 0.48 kB
../src/static/oidc/login.html                 2.60 kB │ gzip: 1.01 kB
../src/static/oidc/assets/style-Bg-wvjxN.css  1.58 kB │ gzip: 0.75 kB
../src/static/oidc/assets/main-DnmqwYeI.js    0.15 kB │ gzip: 0.15 kB
../src/static/oidc/assets/style-Dnfg2NQt.js   0.71 kB │ gzip: 0.40 kB
../src/static/oidc/assets/nested-BvZBmoGC.js  1.06 kB │ gzip: 0.53 kB
✓ built in 140ms
Starting Etherpad...

> etherpad@2.0.3 dev /www/wwwroot/tools/etherpad-lite
> pnpm --filter ep_etherpad-lite run dev "--root"

> ep_etherpad-lite@2.0.3 dev /www/wwwroot/tools/etherpad-lite/src
> node --import tsx node/server.ts "--root"

node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /www/wwwroot/tools/etherpad-lite/src/node/server.ts
    at __node_internal_captureLargerStackTrace (node:internal/errors:496:5)
    at new NodeError (node:internal/errors:405:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:79:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:124:36)
    at defaultLoad (node:internal/modules/esm/load:89:20)
    at nextLoad (node:internal/modules/esm/loader:163:28)
    at ESMLoader.load (node:internal/modules/esm/loader:603:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:457:22)
    at new ModuleJob (node:internal/modules/esm/module_job:64:26)
    at #createModuleJob (node:internal/modules/esm/loader:480:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:434:34) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

Node.js v18.18.2
/www/wwwroot/tools/etherpad-lite/src:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  ep_etherpad-lite@2.0.3 dev: `node --import tsx node/server.ts "--root"`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
root@vmi1855517:/www/wwwroot/tools/etherpad-lite# `

I've checked that TypeScript is installed globally on my system, but it seems like the Node.js environment isn't recognizing the TypeScript files during runtime.

Could anyone provide guidance on how to resolve this issue? Any help would be greatly appreciated!

Thanks in advance.

**Comment by SamTV12345** [comment]
Could you please pull again from develop? I patched support for NodeJS 18 the other day :)

**Comment by mkamranmazhar** [comment, marked as answer]
@SamTV12345 
Thanks it's working now.

Could you please confirm which specific Node.js versions are officially supported by Etherpad? Additionally, if I have the latest version of Node.js installed, would Etherpad work out of the box, or do I need to pull the patched changes from the `develop` branch?

Thanks.

**Reply by SamTV12345** [reply]
You can use NodeJS >= 18 with Etherpad.
